### ISSUE TITLE
Tests behaving strangely with timeout

### ISSUE THREAD

**Issue by deliciouslytyped** [wxWidgets/Phoenix#1316]
See https://github.com/NixOS/nixpkgs/pull/64047/#issuecomment-516108491, what is the correct way to run tests? What is going on with the timeout behaviour?

My suite seems to hang on `unittests/test_lib_agw_piectrl.py::lib_agw_piectrl_Tests::test_lib_agw_piectrlCtor PASSED [ 44%] which is why I enabled timeouts.`which is why I wanted to enable timeouts. (Unless I misinterpreted something earlier)

**Comment by RobinD42** [comment]
Running all the tests en masse has never really worked well. It has to do with trying to use wx in the test cases in a way that it wasn't really meant to be used. Originally it had to do with creating and destroying more than one wx.App as the test cases were run, but even switching to pytests boxed plugin to run each test in a separate process there are still issues related to not running an event loop (I think).  Most of the time running one test module at a time works fine, but trying to run them all will almost certainly have odd failures, or things will lock up and time out.  I expect that there's a way to make things work better, but it hasn't been discovered yet.

**Comment by deliciouslytyped** [comment]
Do you have any idea where the crosstalk is? We could try further isolating components.

**Comment by RobinD42** [comment]
No, it's been a while since I dug into it. I don't recall most of the details at the moment.

**Comment by deliciouslytyped** [comment]
I might take a shot at namespacing the tests, is there some way I can get a list of tests and call them individually?

**Comment by deliciouslytyped** [comment]
By the way, I misunderstood originally, it seems tests are indeed already --boxed by default (a feature of pytest-xdist)
